src/main/java/com/artipie/api/KeyStore.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.api;
+
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.blocking.BlockingStorage;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PfxOptions;
+import java.util.List;
+
+/**
+ * Key store.
+ * @since 0.26
+ */
+public abstract class KeyStore {
```

Comment:
Think we should create an interface `KeyStore` and abstract class `YamlBasedKeyStore`
and so on.